validator/proposer/service.go

```diff
@@ -111,6 +219,27 @@ func (p *Proposer) run(done <-chan struct{}, client pb.ProposerServiceClient) {
 				continue
 			}
 			log.Infof("Block proposed successfully with hash 0x%x", res.BlockHash)
+			p.blockMapping[block.GetSlotNumber()] = nil
+
+		case msg := <-p.attestationBuf:
+			data, ok := msg.Data.(*shardingp2p.AttestationBroadcast)
+			if !ok {
+				log.Error("Received malformed attestation p2p message")
+				continue
+			}
+
+			attestationRecord := data.GetAttestationRecord()
+			block, err := p.GetBlockFromMemory(attestationRecord.GetSlot())
```

Comment:
Looks like this is fetching a block of the same slot as the attestation, but that's not correct. Blocks include attestations for previous slots.